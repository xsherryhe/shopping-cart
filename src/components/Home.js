import { Link } from 'react-router-dom';
import '../styles/Home.css';

export default function Home() {
  return (
    <div className="home">
      <div className="container">
        <h1>The BoardWalk</h1>
        <h2>Board games everywhere you walk!</h2>
        <div className="info">
          <div className="address">
            <h3>Our Address</h3>
            <p>123 Happy Games Lane</p>
            <p>World, Universe 12345</p>
          </div>
          <div className="hours">
            <h3>Our Hours</h3>
            <p>Tuesday - Friday 12pm - 9pm</p>
            <p>Saturday - Sunday 10am - 10pm</p>
          </div>
        </div>
        <Link to="shop">Walk Through Our Online Shop</Link>
      </div>
    </div>
  );
}
