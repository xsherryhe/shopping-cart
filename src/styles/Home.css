.home {
  padding: 10vh 0;
  display: flex;
  justify-content: center;
}

.home .container {
  padding: 3rem;
  background-color: white;
  border-radius: 0.2rem;
  box-shadow: 0 0 10px var(--secondary-color);
}

.home h1 {
  font-size: 4rem;
}

.home h1, .home h2 {
  margin: 0;
}

.home .info {
  display: flex;
  justify-content: center;
  gap: 3rem;
  padding: 2rem 0;
}

.home .info > * {
  border: 3px solid var(--background-color);
  border-radius: 0.2rem;
  padding: 0.5rem 1rem 1rem;
  box-shadow: 0 0 5px var(--secondary-color);
}

.home .info h3 {
  margin: 0.5rem;
}

.home a {
  font-size: 1.2rem;
  color: var(--secondary-color);
}
