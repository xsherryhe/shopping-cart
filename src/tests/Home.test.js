import { render } from '@testing-library/react';
import Home from '../components/Home';

jest.mock('react-router-dom', () => ({
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

describe('Home', () => {
  describe('structure', () => {
    it('renders the home content', () => {
      const { container } = render(<Home />);
      expect(container).toMatchSnapshot();
    });
  });
});
