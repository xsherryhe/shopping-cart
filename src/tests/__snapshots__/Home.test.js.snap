// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home structure renders the home content 1`] = `
<div>
  <div
    class="home"
  >
    <h1>
      The BoardWalk
    </h1>
    <h2>
      Board games everywhere you walk!
    </h2>
    <div
      class="info"
    >
      <div
        class="address"
      >
        <h3>
          Our Address
        </h3>
        <p>
          123 Happy Games Lane
        </p>
        <p>
          World, Universe 12345
        </p>
      </div>
      <div
        class="hours"
      >
        <h3>
          Our Hours
        </h3>
        <p>
          Tuesday - Friday 12pm - 9pm
        </p>
        <p>
          Saturday - Sunday 10am - 10pm
        </p>
      </div>
    </div>
    <a
      href="shop"
    >
      Walk Through Our Online Shop
    </a>
  </div>
</div>
`;
